systems/trajectory_optimization/dircon_position_data.cc

```diff
@@ -56,45 +66,56 @@ void DirconPositionData<T>::updateConstraint(const Context<T>& context) {
           world, world).tail(3);
 
   if (isXZ_) {
-    this->c_ = TXZ_*pt_transform;
-    this->J_ = TXZ_*J3d;
-    this->Jdotv_ = TXZ_*J3d_times_v;
+    this->c_ = TXZ_and_ground_incline_ * pt_transform;
+    this->J_ = TXZ_and_ground_incline_ * J3d;
+    this->Jdotv_ = TXZ_and_ground_incline_ * J3d_times_v;
   } else {
-    this->c_ = pt_transform;
-    this->J_ = J3d;
-    this->Jdotv_ = J3d_times_v;
+    this->c_ = T_ground_incline_ * pt_transform;
+    this->J_ = T_ground_incline_ * J3d;
+    this->Jdotv_ = T_ground_incline_ * J3d_times_v;
   }
   this->cdot_ = this->J_*v;
 }
 
 template <typename T>
-void DirconPositionData<T>::addFixedNormalFrictionConstraints(Vector3d normal,
-                                                              double mu) {
+void DirconPositionData<T>::addFixedNormalFrictionConstraints(double mu) {
   if (isXZ_) {
-    // specifically builds the basis for the x-axis
-    Vector2d normal_xz, d_xz;
-    double L = sqrt(normal(0)*normal(0) + normal(2)*normal(2));
-    normal_xz << normal(0)/L, normal(2)/L;
-    d_xz << -normal_xz(1), normal_xz(0);
+    // builds the basis
+    Vector2d normal_xz(0,1);
```

Comment:
Related to earlier comment re: normal and isXZ. This looks like a change where, previously, we supported non-flat normal vectors for planar contact. Is this no longer the case? If not, why are we removing this functionality?

Reply:
Currently there are c, cdot and cddot in `dircon_kinemaitcs_data`. We want c=0 in the normal direction of the surface, so we rotate the contact position into the surface frame. (I guess we can generalize this more by adding a translation, if there is a specific position where we want the robot to touch the surface? Currently the surface has to contain the global origin.)

Since we rotate c, cdot and cddot because of the above reason, we don't need to rotate it again in the `addFixedNormalFrictionConstraints()`.